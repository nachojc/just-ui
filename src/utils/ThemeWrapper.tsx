import { ReactNode } from "react";
import { ThemeProvider } from "styled-components";
import theme from "../styles/themes";

type ThemeName = keyof typeof theme;

type options = {
  themeName: ThemeName;
};
export const wrapWithTheme = (children: ReactNode, options?: options) => {
  const opt: options = { ...{ themeName: "defaultTheme" }, ...options };
  return <ThemeProvider theme={theme[opt.themeName]}>{children}</ThemeProvider>;
};
