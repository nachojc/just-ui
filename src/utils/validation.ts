export const hasText = (value?: string): boolean => {
  return typeof value === "string" && value?.trim().length > 0;
};
