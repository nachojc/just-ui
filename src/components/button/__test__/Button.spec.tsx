import { render } from "@testing-library/react";
import { create } from "react-test-renderer";
import { Button } from "../";
import React from "react";
import { wrapWithTheme } from "../../../utils/ThemeWrapper";

describe("Button", () => {
  it("render", async () => {
    const btn = render(wrapWithTheme(<Button />));
    expect(btn).toBeTruthy();
  });
  it("render with another Theme", async () => {
    const btn = render(wrapWithTheme(<Button />, { themeName: "publicTheme" }));
    expect(btn).toBeTruthy();
  });
  it("render primary small", async () => {
    const btn = create(wrapWithTheme(<Button variant="primary" size="small"/>));

    expect(btn.toJSON()).toMatchSnapshot();
  });
  it("render tertiary large", async () => {
    const btn = create(wrapWithTheme(<Button variant="tertiary" size="large"/>));

    expect(btn.toJSON()).toMatchSnapshot();
  });
  it("render disabled", async () => {
    const btn = create(wrapWithTheme(<Button disabled/>));

    expect(btn.toJSON()).toMatchSnapshot();
  });
});
