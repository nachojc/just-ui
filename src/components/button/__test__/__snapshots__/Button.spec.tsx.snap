// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render disabled 1`] = `
<DocumentFragment>
  <button
    class="sc-blHHSb cSkiXb"
    disabled=""
    variant="secondary"
  />
</DocumentFragment>
`;

exports[`Button render primary small 1`] = `
<DocumentFragment>
  <button
    class="sc-blHHSb dCeyxJ"
    variant="primary"
  />
</DocumentFragment>
`;

exports[`Button render tertiary large 1`] = `
<DocumentFragment>
  <button
    class="sc-blHHSb eZapBy"
    variant="tertiary"
  />
</DocumentFragment>
`;
